File: app/src/main/java/com/zhai/mvpdemo/AppModule.java
package com.zhai.mvpdemo;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by zhai on 16/6/20.
 */
@Module
public class AppModule {
    private final MyApplication application;

    public AppModule(MyApplication application) {
        this.application = application;
    }

    @Provides
    @Singleton
    Application providesApplication() {
        return application;
    }

    @Provides
    @Singleton
    Context providesContext() {
        return application.getBaseContext();
    }
}

File: app/src/main/java/com/zhai/mvpdemo/AppModule.kt
package com.zhai.mvpdemo

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Named
import javax.inject.Singleton

/**
 * Created by zhai on 16/6/20.
 */
@Module
class AppModule(private val application: MyApplication) {

    @Provides
    @Singleton
    @Named("application")
    internal fun providesApplication(): Application {
        return application
    }

    @Provides
    @Singleton
    @Named("context")
    internal fun providesContext(): Context {
        return application.baseContext
    }
}
